src/CoreWCF.Primitives/src/CoreWCF/Security/SecurityHeader.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.Globalization;
+using CoreWCF.Channels;
+using CoreWCF.Description;
+
+namespace CoreWCF.Security
+{
+    internal abstract class SecurityHeader : MessageHeader
+    {
+        private readonly string actor;
+        private readonly SecurityAlgorithmSuite algorithmSuite;
+        private bool encryptedKeyContainsReferenceList = true;
+        private Message message;
+        private readonly bool mustUnderstand;
+        private readonly bool relay;
+        private bool requireMessageProtection = true;
+        private bool processingStarted;
+        private bool maintainSignatureConfirmationState;
+        private readonly SecurityStandardsManager standardsManager;
+        private MessageDirection transferDirection;
+        private SecurityHeaderLayout layout = SecurityHeaderLayout.Strict;
+
+        public SecurityHeader(Message message,
+            string actor, bool mustUnderstand, bool relay,
+            SecurityStandardsManager standardsManager
+            , SecurityAlgorithmSuite algorithmSuite,
+            MessageDirection transferDirection)
+        {
+            if (message == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("message");
+            }
+            if (actor == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("actor");
+            }
+            if (standardsManager == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("standardsManager");
+            }
+            if (algorithmSuite == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("algorithmSuite");
+            }
+
+            this.message = message;
+            this.actor = actor;
+            this.mustUnderstand = mustUnderstand;
+            this.relay = relay;
+            this.standardsManager = standardsManager;
+            this.algorithmSuite = algorithmSuite;
+            this.transferDirection = transferDirection;
+        }
+
+        public override string Actor
+        {
+            get { return this.actor; }
```

Comment:
Convert to auto property. VS has a helper for this, but it appears on the field definition at the top.